### DISCUSSION TITLE
Use the new `PageStyle` datatype instead of page style string literals

### DISCUSSION THREAD

**Discussion by rvanbekkum** [StefanMaron/BusinessCentral.LinterCop#805] [New Rule]
If runtime 14.0 or newer is used (BC25.0 or newer), then check for any occurences of string literals `'Unfavorable'`, `'Favorable'`, ... (and all other supported page styles).
Raise a diagnostic on the string literal and suggest to use the new `PageStyle` datatype instead (e.g., `PageStyle::Unfavorable`).
Also raise the diagnostic for `Label` variables like `Label 'Unfavorable', Locked = true`

Adopting the use of the new `PageStyle` datatype allows to more easily get the supported pagestyles via IntelliSense and avoids incorrect behaviour when a typo is made in hardcoded strings or label variables.
Unfortunately the `StyleExpr` property on page controls still only accepts strings (or booleans), so a conversion to a string will still be needed, e.g., `Format(MyPageStyleVariable)`

Content of this idea might also be good for alguidelines.dev

**Comment by Arthurvdv** [comment]
Great idea! I've created this rule, but I'm unsure if it will result in many false positives, particularly due to the PageStyle values 'Standard', 'Strong' and 'None.' 

![image](https://github.com/user-attachments/assets/4124d7dc-5241-499a-8d1e-87ae74d663ab)

Validating whether the string literal is consistently passed to a StyleExpr property of a field seems challenging. What do you think?

**Reply by rvanbekkum** [reply]
I think the occurrences of these strings for something else are very low.
The rule shouldn't raise warnings on Captions though, e.g., `Caption = 'Standard'` should not raise a diagnostic (and captions don't need to be checked whatsoever actually).

Should raise a diagnostic:
```al
SomeVariable: Label 'Standard', Locked = true`;
```
```al
MyTextVariable := 'Standard';
```

I think these are very likely to be used as Style and chances of them being used for something else are very low (and for the very rare occurrences, suppressing it with `#pragma warning disable ...` should be a simple resolution).

Other option is skipping these specific strings (Standard, None and Strong). But I think that the earlier mentioned option might be better.

**Reply by Arthurvdv** [reply]
Thank for the feedback on this! With your suggestions added it seems that there are no false positives (at least I didn’t encounter them en there aren’t any reported).

**Comment by Arthurvdv** [comment]
The rule is now available in the release version [v0.32.0](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.32.0) of the LinterCop.
